### DISCUSSION TITLE
Get configuration file change event

### DISCUSSION THREAD

**Discussion by rawvoid** [quarkusio/quarkus#31533] [Q&A]
Hi, guys

I customized an extension and added `ShutdownTask` in `ShutdownContext`. Since the extension starts slowly, in development mode, frequent code changes cause the extension to restart frequently, which takes a lot of time. I only want to execute my `ShutdownTask` when the configuration file changed. How do I get configuration file change events?

Thanks.

**Comment by rawvoid** [comment]
Will Arc reinitialize when quarkus is hot loaded?

**Reply by maxandersen** [reply]
Yes. Extensions are meant to be fast :) 

What is the slow thing doing? Is it Maybe something more aoripriate to put In a devservice?

**Reply by rawvoid** [reply]
> Yes. Extensions are meant to be fast :)
> 
> What is the slow thing doing? Is it Maybe something more aoripriate to put In a devservice?

Thanks, my guess has been verified. Arc will be reinitialized on hot reload, so extended beans cannot be preserved. My extension need refactor.

**Comment by rawvoid** [comment]
Hi, @maxandersen 

Can you tell me how to get config file change event in quarkus extension? Thanks.
